mono/metadata/register-icall-def.h

```diff
@@ -0,0 +1,357 @@
+/**
+ * \file
+ * Copyright 2018 Microsoft
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+// FIXME This belong here:
+// #include "mono/mini/jit-icalls.h"
+// but that requires some factoring.
+//
+// This file is is misnamed.
+// It is not "register" or "def".
+
+#ifndef __MONO_METADATA_REGISTER_JIT_ICALL_DEF_H__
+#define __MONO_METADATA_REGISTER_JIT_ICALL_DEF_H__
+
+// Changes to this file affect AOT file format.
+#define MONO_JIT_ICALLS \
+	\
+MONO_JIT_ICALL (mono_jit_icall_zero_is_reserved)	\
+/* generic_trampoline_* must be first, after zero */	\
+/* These must be ordered like MonoTrampolineType. */	\
+MONO_JIT_ICALL (generic_trampoline_jit)	\
+MONO_JIT_ICALL (generic_trampoline_jump)	\
+MONO_JIT_ICALL (generic_trampoline_rgctx_lazy_fetch)	\
+MONO_JIT_ICALL (generic_trampoline_aot)	\
+MONO_JIT_ICALL (generic_trampoline_aot_plt)	\
+MONO_JIT_ICALL (generic_trampoline_delegate)	\
+MONO_JIT_ICALL (generic_trampoline_generic_virtual_remoting)	\
+MONO_JIT_ICALL (generic_trampoline_vcall)	\
+	\
+/* These must be ordered like MonoTlsKey. */ \
+MONO_JIT_ICALL (mono_tls_get_thread) \
+MONO_JIT_ICALL (mono_tls_get_jit_tls) \
+MONO_JIT_ICALL (mono_tls_get_domain) \
+MONO_JIT_ICALL (mono_tls_get_sgen_thread_info) \
+MONO_JIT_ICALL (mono_tls_get_lmf_addr) \
+	\
+MONO_JIT_ICALL (__emul_fadd)	\
```

Comment:
The `__emul` entries are not subject to ifdef, and there reasons for and against that.